### PR TITLE
22 login page

### LINKED ISSUE #22: Login Page
**GIVEN** I am a logged out User
**WHEN** I access the Login Page
**AND** I fill in a correct username and password
**THEN** I press the Login Button
**AND** I am redirected to the Home page
**AND** get a message saying “Welcome back, Julie”

**GIVEN** I am a logged out User
**WHEN** I access the Login Page
**AND** I fill in the username and password incorrectly
**THEN** I press the Login Button
**AND** I get an error message saying “Username and/or Password do no match”

### PR DESCRIPTION
**Pull request by mattwr18**
#### Issue addressed

fixes #22 

#### Screenshots (if appropriate):

screenshot at 2019-01-11 19 16 16

#### Testing

have not added integration tests, but the coverage is looking good

### CODE REVIEW

**Comment by mattwr18** [comment]
Not sure why it's failing the circleci build with permission denied...

File: package.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
   },
   "dependencies": {
     "axios": "^0.18.0",
+    "izitoast": "^1.4.0",
```
Comment (joaopapereira) [inline package.json]: Do we need to vendorize this library?

File: src/tests/containers/LogIn.test.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+  const emailInput = wrapper.find('input').filterWhere(item => {
+    return item.prop('name') === 'email'
+  })
+
+  const passwordInput = wrapper.find('input').filterWhere(item => {
+    return item.prop('name') === 'password'
+  })
+
+  it('should have a LogIn Header', () => {
```
Comment (joaopapereira) [inline src/tests/containers/LogIn.test.js]: Not sure how valuable is this test

File: src/tests/containers/LogIn.test.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    const mainHeader = wrapper.find('Header').filterWhere(item => {
+      return item.text() === 'Log In'
+    })
+    expect(mainHeader).toHaveLength(1)
+  })
+
+  it('should have a Form', () => {
```
Comment (joaopapereira) [inline src/tests/containers/LogIn.test.js]: Not sure how valuable is this test

File: src/tests/containers/LogIn.test.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  })
+
+  it('should update the state when password input value changes', () => {
+    passwordInput.simulate('change', {
+      target: { value: 'password' }
+    })
+    expect(wrapper.state().password).toBe('password')
+  })
+
+  it('should call handleLogIn and postLogInInfo when the form is submitted', async () => {
```
Comment (joaopapereira) [inline src/tests/containers/LogIn.test.js]: I would like to see the negative test

File: src/tests/containers/LogIn.test.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+      return item.text() === 'Log In'
+    })
+    expect(mainHeader).toHaveLength(1)
+  })
+
+  it('should have a Form', () => {
+    expect(wrapper.find('Form')).toHaveLength(1)
+  })
+
+  it('should update the state when email input value changes', () => {
```
Comment (joaopapereira) [inline src/tests/containers/LogIn.test.js]: Not sure how valuable is this test

File: src/tests/containers/LogIn.test.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    emailInput.simulate('change', {
+      target: { value: 'existing-user@example.com' }
+    })
+    expect(wrapper.state().email).toBe('existing-user@example.com')
+  })
+
+  it('should update the state when password input value changes', () => {
```
Comment (joaopapereira) [inline src/tests/containers/LogIn.test.js]: Not sure how valuable is this test

File: src/tests/reducers/usersReducer.test.js [hunk 7]
```diff
@@ -13,5 +13,11 @@ describe('reduces users', () => {
         payload: ['User to be added to store']
       })
     ).toEqual(['User to be added to store'])
+    expect(
```
Comment (joaopapereira) [inline src/tests/reducers/usersReducer.test.js]: Shouldn't this be in a separate it block?

File: src/containers/LogIn.js [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+  handleLogIn = async e => {
+    const { email, password } = this.state
+    e.preventDefault()
+    await this.props
+      .postLogInInfo({ email, password })
+      .then(() => {
+        iziToast.show({
+          theme: 'light',
+          title: 'Success',
+          message: 'Have a look around',
```
Comment (joaopapereira) [inline src/containers/LogIn.js]: The text on the story is:
"Welcome back, Julie"

**Review by joaopapereira — CHANGES_REQUESTED** [review]
Good job getting this together. I have some comments and questions
Please review them

File: package.json (hunk 1 above)
Comment (mattwr18) [reply package.json]: we use this for our toastr notifications on successful or unsuccessful login, so we don't need it, but I think it is quite nice

Comment (joaopapereira) [reply package.json]: Not questioning the use of the library. What I am questioning is why we have it in the package.jsno and at the same time we add the minified JavaScript and css of it to our code.

I prefer not to vendorize anything

Comment (mattwr18) [reply package.json]: oh, right, sorry I misunderstood, and we totally can remove that since we are importing it from our assets directory

Comment (joaopapereira) [reply package.json]: Is there a way to use the package.json version instead of the assets folder one?

**Pull request merged**